src/Npgsql/ThrowHelper.cs

```diff
@@ -7,11 +7,11 @@ namespace Npgsql
     static class ThrowHelper
     {
         [DoesNotReturn]
-        internal static void ThrowNotSupportedException_NoPropertyGetter(Type type, MemberInfo property) =>
-            throw new NotSupportedException($"Composite type {type} cannot be written because the {property} property has no getter.");
+        internal static void ThrowMissingMemberException_NoPropertyGetter(Type type, MemberInfo property) =>
+            throw new MissingMemberException($"Composite type {type} cannot be written because the {property} property has no getter.");
```

Comment:
I'm not sure we should be throwing reflection-specific exception types such as MissingMemberException. This could confuse users which don't understand or know the connection between this and our composite handling (any use of reflection by us is an internal implementation detail). 

On the other hand NotSupportedException wasn't very good either.

FYI EF Core have made the decision to throw InvalidOperationException for (almost?) all errors caused by incorrect user use of the APIs. 

Bottom line, I guess I'm OK with this.

Reply:
Not happy with this too. Searched all corefx repo and found that only `Lazy<T>` uses this exception type, so... I reread the documentation about `InvalidOperationException` and it looks correct for this case.